### PR TITLE
add support for --verify-easyconfig-filenames

### PR DESCRIPTION
**Pull request by boegel**
by popular request...

cc @shahzebsiddiqui, @damianam, @pneerincx

### CODE REVIEW

File: easybuild/framework/easyconfig/easyconfig.py [hunk 1]
```diff
@@ -1389,7 +1389,7 @@ def verify_easyconfig_filename(path, specs, parsed_ec=None):
     :param parsed_ec: (list of) EasyConfig instance(s) corresponding to easyconfig file
     """
     if isinstance(parsed_ec, EasyConfig):
-        ecs = [parsed_ec]
+        ecs = [{'ec': parsed_ec}]
```
Comment (wpoely86) [inline easybuild/framework/easyconfig/easyconfig.py]: is this a bug fix?

File: easybuild/tools/options.py [hunk 2]
```diff
@@ -383,9 +384,10 @@ def override_options(self):
                              str, 'store', False, {'metavar': "PATH"}),
             'use-existing-modules': ("Use existing modules when resolving dependencies with minimal toolchains",
                                      None, 'store_true', False),
+            'verify-easyconfig-filenames': ("Verify whether filename of specified easyconfigs matches with contents",
```
Comment (wpoely86) [inline easybuild/tools/options.py]: put it to `True` by default?

File: easybuild/framework/easyconfig/easyconfig.py (hunk 1 above)
Comment (boegel) [inline easybuild/framework/easyconfig/easyconfig.py]: yes, and the enhanced test covers it now

File: easybuild/tools/options.py (hunk 2 above)
Comment (boegel) [inline easybuild/tools/options.py]: No, that would change the behaviour, sites should enable this if they want to stricter behaviour.
I don't consider the flexibility of filenames for specified easyconfigs a bug, but a feature. I'd get annoyed fi EasyBuild gets picky about filenames when it doesn't really matter; it only really matters for the dependency resolver, which already complains if an easyconfig file that it picks up by filename doesn't contain what is expected.

We can reconsider this in EasyBuild v4.0, but I'm in favour of leaving the default behaviour as is.

**Comment by boegel** [comment]
Going in, thanks for the review @wpoely86 and the thumbs up @pneerincx

**Pull request merged**